### ISSUE TITLE
(2.0) Unit and description not taken into account for metrics from producers

### ISSUE THREAD

**Issue by jmartisk** [smallrye/smallrye-metrics#38]
```
@Gauge(unit = "USD", name = "coffee_price", absolute = true)
public long getPrice() {
    return 4;
}

@Produces
@Metric(name = "coffee_price_produces", unit = "USD", absolute = true)
protected Gauge<Long> getPrice2() {
    return () -> 4L;
}
```

These declarations should be equivalent, but the latter doesn't take the unit into account at runtime.
```
curl -H"Accept: application/json" -X OPTIONS localhost:8080/metrics/application/coffee_price_produces
{
    "coffee_price_produces":{
        "unit":"none",
        "type":"gauge",
        "displayName":"coffee_price_produces",
        "tags":""
    }
```

**Comment by jmartisk** [comment]
Looks like the description is not picked up either.

**Comment by jmartisk** [comment]
pull request for this for 1.1 branch: https://github.com/smallrye/smallrye-metrics/pull/49 
TCK test to cover this: https://github.com/eclipse/microprofile-metrics/pull/314

**Comment by jmartisk** [comment]
Fixes for 2.0:
TCK: sent PR to specification master https://github.com/eclipse/microprofile-metrics/pull/351
SmallRye: for the time being the fix is in my 2.0 branch https://github.com/jmartisk/smallrye-metrics/commits/tags-impl (commit 40c7287ff779963e91f26ce06031a46808cd281f)

**Comment by jmartisk** [comment]
Merged for 2.0
